app/routers/google_connect.py

```diff
@@ -0,0 +1,261 @@
+                           ) -> OAuthCredentials:
+
+    oauth_credentials = OAuthCredentials(
+        owner=user,
+        token=credentials.token,
+        refresh_token=credentials.refresh_token,
+        token_uri=credentials.token_uri,
+        client_id=credentials.client_id,
+        client_secret=credentials.client_secret,
+        expiry=credentials.expiry
+    )
+
+    session.add(oauth_credentials)
+    session.commit()
+    return credentials
+
+
+def get_active_user(session: SessionLocal = Depends()) -> User:
+    # TODO - get connected/current user
+    user = session.query(User).filter_by(id=1).first()
+    return user
+
+
+def is_client_secret_none() -> bool:
+    return CLIENT_SECRET_FILE is None
+
+
+def get_current_year_events(
+        credentials: Credentials, user: User,
+        session: SessionLocal = Depends()) -> list:
+    '''Getting user events from google calendar'''
+
+    currrnt_year = datetime.now().year
+    start = datetime(currrnt_year, 1, 1).isoformat() + 'Z'
+    end = datetime(currrnt_year + 1, 1, 1).isoformat() + 'Z'
+
+    service = build('calendar', 'v3', credentials=credentials)
+    events_result = service.events().list(
+        calendarId='primary',
+        timeMin=start,
+        timeMax=end,
+        singleEvents=True,
+        orderBy='startTime'
+    ).execute()
+
+    events = events_result.get('items', [])
+    return events
+
+
+def push_events_to_db(events: list, user: User,
+                      session: SessionLocal = Depends()) -> bool:
+    '''Adding google events to db'''
+    db_cleanup(user, session)
+
+    for event in events:
+        # Running over the events that have come from the API
+        location = None
+        title = event['summary']  # The Google event title
+
+        # support for all day events
+        if 'dateTime' in event['start'].keys():
+            # This case handles part time events (not all day events)
+            start = datetime.fromisoformat(event['start']['dateTime'])
+            end = datetime.fromisoformat(event['end']['dateTime'])
+        else:
+            # This case handles all day events
+            # It takes the date string in this format yyyy-mm-dd
+            # - according to the documentation
+            # and then split it and passing it to datetime constructor.
+            start_in_str = event['start']['date'].split('-')
+            start = datetime(
+                year=int(start_in_str[0]),
+                month=int(start_in_str[1]),
+                day=int(start_in_str[2])
+            )
+
+            end_in_str = event['end']['date'].split('-')
+            end = datetime(
+                year=int(end_in_str[0]),
+                month=int(end_in_str[1]),
+                day=int(end_in_str[2])
+            )
+
+        if 'location' in event.keys():
+            # This case handles if Google Event has a location attached
+            location = event['location']
```

Comment:
Prefer `location = event.get('location')`, without the `if`

Reply:
Done.